### ISSUE TITLE
Method for scrolling to a time

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#467]
Originally reported on Google Code with ID 196

```
Is there any way to go to a particular time in the agenda view once the
calendar has been initialized?
```

Reported by `lfrancop` on 2009-11-18 09:34:51

Imported with **14** stars.

**Comment by laander** [comment]
+1 for this, could really use this, especially since we're loading events after calendar load and then injecting them. If some events are outside the current hour view in agenda, there's no indication that you have to scroll and users get a bad experience.

**Comment by arshaw** [comment]
these improvements would also apply to timeline view, or any other view that has a chronological scroll.

**Comment by tlays11** [comment]
+1 for the timeline view scrolling.

**Comment by arshaw** [comment]
would want the ability to align the specified datetime either left (or right if RTL) or centered.

**Comment by evanjmg** [comment]
+1 -  Here's my quick solution for now - It's very hard to do it  jquery and it's a total hack - had to do something like this to center the event 
```
  let offsetEvent = ($(`a.fc-timeline-event#${event.id}`).offset().left);

    $(`tbody .fc-time-area .fc-scroller`).animate({
        scrollLeft: '-=' + ($(`tbody .fc-time-area .fc-scroller`).width()  / 2  -  offsetEvent)
    }, 600);
```

**Comment by laander** [comment]
Here's how we ended up implementing it for agenda view in booking.js: https://github.com/timekit-io/booking-js/blob/master/src/main.js#L199

It's quite cumbersome, but takes a lot of scenarios into account:
- Take the `minTime` setting into account
- Take the `slotDuration` into account
- Only scroll if event is outside current viewport

Thought somebody might be interested in it until official support arrives 👌

**Comment by mmccallum** [comment]
Any movement on the ability to scroll to a time using gotoDate?  Appears that it was accepted last year, then closed recently without being implemented.

> could be with `gotoDate`, which accepts a date AND time
> or maybe 'scrollToTime`, which only accepts a time
> Reported by adamrshaw on 2013-08-14 03:40:04

**Comment by acerix** [comment]
This issue is still open, please **give a** 👍  **to vote** ([?](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments)), press the 🔈**Subscribe** button to receive updates.

If you are referring to https://github.com/fullcalendar/fullcalendar/issues/3878, that was closed because it is a duplicate of this one.

**Comment by guilhermefalcao** [comment]
+1

**Comment by korengal** [comment]
any news ?

**Comment by jamesamuel** [comment]
What's the status on this?

**Comment by 5angel** [comment]
3 years for a simple param? You gotta be kidding me

**Comment by vgb0332** [comment]
For real, I thought it would have been added or solved officially by now, but no luck I guess. Personally, I addded the now indicator on the calendar and look for the dom element myself and scroll to that position manually. This works but it would be better if legit method comes out officially.

**Comment by maxcr** [comment]
Uh its built into the HTML spec. ScrollIntoView

```
        $(sel)[0].scrollIntoView(true, {
          behavior: 'smooth',
          block: 'start'
        })
``

**Comment by jonasgroendahl** [comment]
Is this coming in v4?

**Comment by arshaw** [comment]
this is released in [v4.1.0](https://github.com/fullcalendar/fullcalendar/releases/tag/v4.1.0)

https://fullcalendar.io/docs/scrollToTime

sorry this took almost 10 years

**Comment by Konstantin-Ushakov** [comment]
What about to scroll a date with specific time?
The code that should help didn't work:
```
const calendarApi = calendar.getApi();
const date = moment('2022-12-12T12:00:00+00:00', 'yyyy-MM-DDTHH:mm:ssZ');
calendarApi?.gotoDate(date.format('yyyy-MM-DDTHH:00:00Z'));
calendarApi?.scrollToTime(date.format('yyyy-MM-DDTHH:00:00Z'));
```
It scroll to 00:00 of day...
The same behaviour are:
```
calendarApi?.scrollToTime(date.format('yyyy-MM-DDTHH:00:00Z'));
calendarApi?.gotoDate(date.format('yyyy-MM-DDTHH:00:00Z'));
```
```
calendarApi?.gotoDate(date.format('yyyy-MM-DD'));
calendarApi?.scrollToTime(date.format('HH:00:00Z'));
```
The only code is help is connected with states:
```
const initialScrollToDateTime = (props.scrollDateTime
    ? moment(props.scrollDateTime)
    : moment().set({
      hour: 0,
      minute: 0,
      second: 0,
      millisecond: 0,
    })
  );
  const [scrollInfo, setScrollInfo] = useState<{
    dateTime: Moment,
    scrolledDate: boolean,
    scrolledTime: boolean,
  }>({
    dateTime: initialScrollToDateTime,
    scrolledDate: false,
    scrolledTime: false,
  });
...
  useEffect(() => {
    if (props.scrollDateTime) {
      const scrollDateTimeFormatted = scrollInfo.dateTime.format('yyyy-MM-DDTHH:mm:ssZ');
      const propsScrollDateTimeFormatted = props.scrollDateTime.format('yyyy-MM-DDTHH:mm:ssZ');
      if (scrollDateTimeFormatted !== propsScrollDateTimeFormatted) {
        setScrollInfo({
          dateTime: props.scrollDateTime,
          scrolledDate: false,
          scrolledTime: false,
        });
      }
    }
  }, [props.scrollDateTime]);

  useEffect(() => {
    if (scrollInfo.dateTime && calendarRef) {
      const calendarApi = calendarRef.current?.getApi();
      if (calendarApi) {
        const date = scrollInfo.dateTime;
        if (!scrollInfo.scrolledDate) {
          calendarApi?.gotoDate(date.format('yyyy-MM-DD'));
        } else if (!scrollInfo.scrolledTime) {
          calendarApi?.scrollToTime(date.format('HH:00:00Z'));
        }
      }
    }
  }, [scrollInfo.scrolledDate, scrollInfo.scrolledTime, scrollInfo.dateTime.format('yyyy-MM-DDTHH:mm:ssZ')]);

  useEffect(() => {
    if (scrollInfo.dateTime && calendarRef) {
      const calendarApi = calendarRef.current?.getApi();
      if (calendarApi) {
        const date = scrollInfo.dateTime;
        const momentCurrentDate = moment(calendarApi.getDate().toUTCString());
        if (momentCurrentDate.format('yyyy-MM-DD') === date.format('yyyy-MM-DD')) {
          setScrollInfo({
            ...scrollInfo,
            scrolledDate: true,
          });
        } else if (momentCurrentDate.format('HH:00:00Z') === date.format('HH:00:00Z')) {
          setScrollInfo({
            ...scrollInfo,
            scrolledTime: true,
          });
        }
      }
    }
  }, [scrollInfo.scrolledDate, scrollInfo.scrolledTime, scrollInfo.dateTime.format('yyyy-MM-DDTHH:mm:ssZ')]);
```
